script/draft-release/version.ts

```diff
@@ -0,0 +1,57 @@
+import { inc, parse } from 'semver'
+
+import { Channel } from './channel'
+
+export function getNextVersionNumber(
+  version: string,
+  channel: Channel
+): string {
+  const parsed = parse(version)
+
+  if (parsed == null) {
+    throw new Error(`Unable to parse input '${version}' into version`)
+  }
+
+  if (channel === 'production') {
```

Comment:
Can we be explicit with `channel` and check for all possible values exhaustively. Conditionals work, but I think a switch makes it that much clearer.

```ts
  switch (channel) {
    case 'production': break
    case 'beta': break
    case 'test': break
    case 'development': break
  }
```